chromium_src/components/history/core/browser/sync/typed_url_sync_bridge.cc

```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "components/history/core/browser/sync/typed_url_sync_bridge.h"
+
+#include "brave/components/brave_sync/features.h"
+#include "ui/base/page_transition_types.h"
+
+namespace {
+
+bool IsSendAllHistoryEnabled() {
+  return base::FeatureList::IsEnabled(
+      brave_sync::features::kBraveSyncSendAllHistory);
+}
+
+}  // namespace
+
+namespace ui {
+
+static bool PageTransitionCoreTypeIs_BraveImpl(PageTransition lhs,
+                                               PageTransition rhs) {
+  if (IsSendAllHistoryEnabled() &&
+      PageTransitionCoreTypeIs(rhs, ui::PAGE_TRANSITION_RELOAD)) {
+    // At TypedURLSyncBridge::WriteToTypedUrlSpecifics when syncing all
+    // history, don't ignore reload transitions.
+    return false;
+  }
+  return PageTransitionCoreTypeIs(lhs, rhs);
+}
+
+}  // namespace ui
+
+#define PageTransitionCoreTypeIs PageTransitionCoreTypeIs_BraveImpl
+
+#include "src/components/history/core/browser/sync/typed_url_sync_bridge.cc"
+
+#undef PageTransitionCoreTypeIs
+
+namespace history {
+
+// static
+bool TypedURLSyncBridge::HasTypedUrl(const std::vector<VisitRow>& visits) {
+  if (IsSendAllHistoryEnabled()) {                                  \
+    /* When there are no visits, we must return false. Otherwise */ \
```

Comment:
there're some leftovers at the end of lines :)

Reply:
Sorry, changed at https://github.com/brave/brave-core/pull/16705/commits/5263fa9283511c997491005baa6f92ec348da289